devito/finite_differences/differentiable.py

```diff
@@ -548,20 +548,20 @@ def __new__(cls, *args, **kwargs):
         nested, others = split(args, lambda e: isinstance(e, Mul))
         args = flatten(e.args for e in nested) + list(others)
 
+        # Gather all numbers and simplify
+        nums, others = split(args, lambda e: isinstance(e, (int, float,
+                                                            sympy.Number, np.number)))
+        scalar = sympy.Mul(*nums)
+
         # a*0 -> 0
-        if any(i == 0 for i in args):
+        if scalar == 0:
             return sympy.S.Zero
 
         # a*1 -> a
-        args = [i for i in args if i != 1]
-
-        # a*-1 -> a*-1
-        # a*-1*-1 -> a
-        # a*-1*-1*-1 -> a*-1
-        nminus = len([i for i in args if i == sympy.S.NegativeOne])
-        args = [i for i in args if i != sympy.S.NegativeOne]
-        if nminus % 2 == 1:
-            args.append(sympy.S.NegativeOne)
+        if scalar - 1 == 0:
```

Comment:
elif scalar == 1 ?

Reply:
No this creates issues, for example `Sympy.S.One == 1` is False. Only thing that always compares properly is zero.